pairing_api/src/pairing.rs

```diff
@@ -0,0 +1,472 @@
+        let irn_metadata = params.irn_metadata();
+        let response = Response::Success(SuccessfulResponse {
+            id: message_id,
+            jsonrpc: JSON_RPC_VERSION_STR.into(),
+            result: serde_json::to_value(params)
+                .map_err(|err| PairingClientError::EncodeError(err.to_string()))?,
+        });
+
+        // Publish the encrypted message
+        self.publish_payload(topic, irn_metadata, Payload::Response(response), client)
+            .await?;
+
+        println!("\nOutbound request sent!");
+
+        Ok(())
+    }
+
+    /// Private function to publish a payload.
+    async fn publish_payload(
+        &self,
+        topic: &str,
+        irn_metadata: IrnMetadata,
+        payload: Payload,
+        client: &Client,
+    ) -> Result<(), PairingClientError> {
+        // try to extend session before updating local store.
+        let sym_key = {
+            let pairings = self.pairings.lock().await;
+            let pairing = pairings
+                .get(topic)
+                .ok_or_else(|| PairingClientError::PairingNotFound)?;
+            hex::decode(pairing.sym_key.clone()).map_err(|err| {
+                PairingClientError::EncodeError(format!("Failed to decode sym_key: {:?}", err))
+            })?
+        };
+
+        let payload = serde_json::to_string(&payload)
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+        let message = encrypt_and_encode(EnvelopeType::Type0, payload, &sym_key)
+            .map_err(|e| anyhow::anyhow!(e))
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+
+        // Publish the encrypted message
+        {
+            client
+                .publish(
+                    topic.into(),
+                    message,
+                    None,
+                    irn_metadata.tag,
+                    Duration::from_secs(irn_metadata.ttl),
+                    irn_metadata.prompt,
+                )
+                .await
+                .map_err(PairingClientError::PingError)?;
+        };
+
+        Ok(())
+    }
+
+    /// Private function to generate a WalletConnect URI.
+    fn generate_uri(pairing: &PairingInfo, sym_key: &str) -> String {
+        let methods_str = pairing
+            .methods
+            .0
+            .iter()
+            .map(|method_group| format!("[{}]", method_group.join(",")))
+            .collect::<Vec<_>>()
+            .join(",");
+        let expiry_timestamp_str = pairing.expiry.to_string();
+
+        format!(
+            "wc:{}@{}?symKey={}&methods={}&relay-protocol={}&expiryTimestamp={}",
+            pairing.topic,
+            VERSION,
+            sym_key,
+            methods_str,
+            pairing.relay.protocol,
+            expiry_timestamp_str
+        )
+    }
+}
+
+fn gen_sym_key() -> String {
+    let mut sym_key = [0u8; 32];
+    OsRng.fill_bytes(&mut sym_key);
+    hex::encode(sym_key)
+}
```

Comment:
```rust
fn gen_sym_key() -> String {
    hex::encode(OsRng.gen::<[u8; 32]>())
}
```
also can inline always it if you want

Reply:
done

Reply:
Still there 👀 
https://github.com/KomodoPlatform/WalletConnectRust/blob/37c929d1e20d730414abeee8087b7ecc52cc84a4/pairing_api/src/pairing.rs#L420-L424

Reply:
pushed @ https://github.com/KomodoPlatform/WalletConnectRust/pull/3/commits/71acc1a2c10dc8dba1ea00ad4b78e063ec25fa6c